modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/batch/Discovery.scala

```diff
@@ -90,6 +94,29 @@ object Discovery {
     (incomplete, unshredded)
   }
 
+  /**
+   * Most likely folder names should have "s3://path/run=YYYY-MM-dd-HH-mm-ss" format.
+   * This function extracts timestamp in the folder names and returns the ones in the
+   * given interval. Folders that does not have this format are included in the result also.
+   */
+  def findIntervalFolders(folders: List[Folder], since: Option[Instant], until: Option[Instant]): List[Folder] =
+    (since, until) match {
+      case (None, None) => folders
+      case _ =>
+        val (nonTimeFormatted, inInterval) = folders.foldLeft[(List[Folder], List[Folder])]((Nil, Nil)) {
+          case ((l, r), f) =>
+            val folderName = f.folderName
+            if (folderName.startsWith("run=")) {
+              val time = folderName.stripPrefix("run=")
+              Common.parseFolderTime(time)
+                .fold(_ => (f::l, r), {
+                  case time if since.fold(true)(_.isBefore(time)) && until.fold(true)(_.isAfter(time)) => (l, f::r)
+                  case _ => (l, r)
+                })
```

Comment:
I realized using `filter` is better than `foldLeft` therefore change the function a bit. I think `fold` is going better in here with the new version of this function but let me know if you still think otherwise.